python-checks/src/main/java/org/sonar/python/checks/ArgumentNumberCheck.java

```diff
@@ -92,20 +91,8 @@ private static boolean isException(CallExpression callExpression, FunctionSymbol
   }
 
   private static boolean extendsZopeInterface(@Nullable Symbol symbol) {
-    if (symbol != null && symbol.kind() == Symbol.Kind.CLASS) {
-      Set<Symbol> superClasses = new HashSet<>();
-      addSuperClasses(symbol, superClasses);
-      return superClasses.stream().anyMatch(s -> "zope.interface.Interface".equals(s.fullyQualifiedName()));
-    }
-    return false;
-  }
-
-  private static void addSuperClasses(Symbol symbol, Set<Symbol> set) {
-    if (set.add(symbol) && symbol instanceof ClassSymbol) {
-      for (Symbol superClass : ((ClassSymbol) symbol).superClasses()) {
-        addSuperClasses(superClass, set);
-      }
-    }
+    return SymbolUtils.findAllSuperClasses(symbol).stream().anyMatch(
```

Comment:
I think it's better to use `symbol.isOrExtends("zope.interface.Interface")` here. So I think we'll have to change `isOrExtends` implementation to make this work. I can help you with that